tools/ui/src/auth/auth.ts

```diff
@@ -2,135 +2,145 @@ import { Principal } from "@dfinity/principal"
+  auth!.innerHTML = "";
 
-  CopyId()
-  LogoutButton()
+  CopyId();
+  LogoutButton();
 }
 
 function LogoutButton() {
-  const auth = document.getElementById("authentication")
+  const auth = document.getElementById("authentication");
 
-  const buttonLogout = document.createElement("button")
-  buttonLogout.className = "btn random"
-  buttonLogout.style.margin = "10px"
-  buttonLogout.innerText = "Logout"
+  const buttonLogout = document.createElement("button");
+  buttonLogout.className = "btn random";
+  buttonLogout.style.margin = "10px";
+  buttonLogout.innerText = "Logout";
 
   buttonLogout.addEventListener("click", async () => {
-    await logout()
+    await logout();
   })
 
-  auth!.appendChild(buttonLogout)
+  auth!.appendChild(buttonLogout);
 }
 
 function CopyId() {
   if (!authClient) {
-    return
+    return;
   }
 
-  const auth = document.getElementById("authentication")
+  const auth = document.getElementById("authentication");
 
-  const copyText = document.createElement("span")
+  const copyText = document.createElement("span");
 
-  const id = authClient.getIdentity().getPrincipal().toString()
-  const idShort = id.slice(0, 5) + "..." + id.slice(-5)
-  copyText.innerText = idShort
+  const id = authClient.getIdentity().getPrincipal().toString();
+  const idShort = id.slice(0, 5) + "..." + id.slice(-5);
+  copyText.innerText = idShort;
 
-  const copyButton = document.createElement("button")
-  copyButton.id = "copyButton"
-  copyButton.style.cursor = "pointer"
-  copyButton.innerHTML = `${copyIcon}`
+  const copyButton = document.createElement("button");
+  copyButton.id = "copyButton";
+  copyButton.style.cursor = "pointer";
+  copyButton.innerHTML = `${copyIcon}`;
 
   copyButton.addEventListener("click", function () {
     navigator.clipboard.writeText(id).catch((err) => {
-      console.error(err)
+      console.error(err);
     })
-    copyText.innerText = "Copied!"
+    copyText.innerText = "Copied!";
     setTimeout(() => {
-      copyText.innerText = idShort
+      copyText.innerText = idShort;
     }, 2000)
   })
 
-  auth?.appendChild(copyButton)
-  auth?.appendChild(copyText)
-}
-
-async function login(options: DerivationOriginOptions) {
-  authClient?.login({
-    identityProvider,
-    derivationOrigin: derivationOrigin(options),
-    onSuccess: async () => {
-      refresh_actor(options.canisterId)
-      insertLogout()
-    },
-    onError: (err) => console.error(err),
-  })
+  auth?.appendChild(copyButton);
+  auth?.appendChild(copyText);
 }
 
 async function logout() {
-  authClient?.logout()
-  window.location.reload()
-}
-
-function derivationOrigin(options: DerivationOriginOptions): string {
-  if (options.domain !== "icp0.io" && options.domain !== "ic0.app") {
-    throw new Error("Invalid domain")
-  }
-
-  return options.raw
-    ? "https://" + options.canisterId.toString() + ".raw." + options.domain
-    : "https://" + options.canisterId.toString() + "." + options.domain
+  authClient?.logout();
```

Comment:
`AuthClient.logout()` is apparently an async function ([source](https://github.com/dfinity/agent-js/blob/40d98004cd7cb1da7fe62794762b9ffc24a4d21e/packages/auth-client/src/index.ts#L555)):

```suggestion
  await authClient?.logout();
```